diff --git a/packages/services/Telecomm/src/com/android/server/telecom/CallAudioModeStateMachine.java b/packages/services/Telecomm/src/com/android/server/telecom/CallAudioModeStateMachine.java
index e6ad9d5..e04d004 100644
--- a/packages/services/Telecomm/src/com/android/server/telecom/CallAudioModeStateMachine.java
+++ b/packages/services/Telecomm/src/com/android/server/telecom/CallAudioModeStateMachine.java
@@ -16,6 +16,7 @@
 
 package com.android.server.telecom;
 
+import android.content.Context;
 import android.media.AudioManager;
 import android.os.Message;
 import android.os.SystemProperties;
@@ -23,6 +24,7 @@ import android.telecom.PhoneAccount;
 import android.telecom.PhoneAccountHandle;
 import android.telecom.TelecomManager;
 import android.telephony.SubscriptionManager;
+import android.telephony.SubscriptionInfo;
 import android.telephony.TelephonyManager;
 import android.util.SparseArray;
 
@@ -154,7 +156,12 @@ public class CallAudioModeStateMachine extends StateMachine {
         public void enter() {
             if (mIsInitialized) {
                 Log.i(LOG_TAG, "Abandoning audio focus: now UNFOCUSED");
+                boolean setRealcallAudioParams = SystemProperties
+                    .getBoolean("ro.telephony.samsung.realcall", false);
                 mAudioManager.abandonAudioFocusForCall();
+                if (setRealcallAudioParams) {
+                    mAudioManager.setParameters("realcall=off");
+                }
                 mAudioManager.setMode(AudioManager.MODE_NORMAL);
 
                 mMostRecentMode = AudioManager.MODE_NORMAL;
@@ -295,6 +302,8 @@ public class CallAudioModeStateMachine extends StateMachine {
             Log.i(LOG_TAG, "Audio focus entering SIM CALL state");
             boolean setMsimAudioParams = SystemProperties
                     .getBoolean("ro.multisim.set_audio_params", false);
+            boolean setRealcallAudioParams = SystemProperties
+                    .getBoolean("ro.telephony.samsung.realcall", false);
             Call call = mCallAudioManager.getForegroundCall();
 
             mAudioManager.requestAudioFocusForCall(AudioManager.STREAM_VOICE_CALL,
@@ -306,6 +315,9 @@ public class CallAudioModeStateMachine extends StateMachine {
                 int subId = TelephonyManager.getDefault().getSubIdForPhoneAccount(account);
                 int phoneId = SubscriptionManager.getPhoneId(subId);
                 Log.d(LOG_TAG, "setAudioParameters phoneId=" + phoneId);
+                if(setRealcallAudioParams){
+                    mAudioManager.setParameters("realcall=on");
+                }
                 if (phoneId == 0) {
                     mAudioManager.setParameters("phone_type=cp1");
                 } else if (phoneId == 1) {
